src/resources/nfts/simplehash/utils.ts

```diff
@@ -193,3 +204,206 @@ export function simpleHashNFTToUniqueAsset(nft: SimpleHashNFT): UniqueAsset {
+            ({
+              marketplaceId: getInternalMarketplaceIdFromSimpleHashMarketplaceId(
+                floorPrice.marketplace_id
+              ),
+              paymentToken: {
+                address: floorPrice.payment_token.address,
+                decimals: floorPrice.payment_token.decimals,
+                name: floorPrice.payment_token.name,
+                symbol: floorPrice.payment_token.symbol,
+              },
+              value: floorPrice.value,
+            } as NFTFloorPrice)
+        ),
+      imageUrl: collection.image_url
+        ? maybeSignUri(collection.image_url)
+        : undefined,
+      name:
+        uniqueTokenType === uniqueTokenTypes.ENS
+          ? ENS_COLLECTION_NAME
+          : collection.name ?? undefined,
+      simpleHashSpamScore: collection?.spam_score ?? undefined,
+      twitter: collection.twitter_username ?? undefined,
+    },
+    contract: {
+      address: nft.contract_address,
+      name: nft.contract.name ?? undefined,
+      schema_name: nft.contract.type,
+      symbol: nft.contract.symbol ?? undefined,
+    },
+    description: nft.description ?? undefined,
+    externalUrl: nft.external_url ?? undefined,
+    images: {
+      blurhash: nft.previews?.blurhash ?? undefined,
+      fullResPngUrl,
+      fullResUrl,
+      lowResPngUrl,
+      // mimeType only corresponds to fullResUrl
+      mimeType: nft.image_properties?.mime_type ?? undefined,
+    },
+    isSendable:
+      // can't send poaps because they're on gnosis
+      uniqueTokenType !== uniqueTokenTypes.POAP &&
+      (nft.contract.type === ERC721 || nft.contract.type === ERC1155),
+    lastSale:
+      nft.last_sale?.payment_token?.name &&
+      nft.last_sale?.payment_token?.symbol &&
+      !isNil(nft.last_sale?.unit_price)
+        ? {
+            paymentToken: {
+              address: nft.last_sale?.payment_token.address,
+              decimals: nft.last_sale?.payment_token.decimals,
+              name: nft.last_sale?.payment_token.name,
+              symbol: nft.last_sale?.payment_token.symbol,
+            },
+            value: nft.last_sale?.unit_price,
+          }
+        : undefined,
+    marketplaces: nft.collection.marketplace_pages
+      .filter(
+        (marketplace: SimpleHashMarketplace) =>
+          // filter out unsupported marketplaces
+          !!getInternalMarketplaceIdFromSimpleHashMarketplaceId(
+            marketplace.marketplace_id
+          )
+      )
+      .map(
+        (marketplace: SimpleHashMarketplace) =>
+          ({
+            collectionId: marketplace?.marketplace_collection_id,
+            collectionUrl: marketplace?.collection_url,
+            marketplaceId: getInternalMarketplaceIdFromSimpleHashMarketplaceId(
+              marketplace.marketplace_id
+            ),
+            name: marketplace?.marketplace_name,
+            nftUrl: marketplace?.nft_url,
+          } as NFTMarketplace)
+      ),
+    name: nft.name ?? undefined,
```

Comment:
How could an NFT not have a name?

Reply:
bc a `SimpleHashNFT` defines `name: string | undefined`. I put a note in the docblock saying that this function may produce an invalid NFT if `filterSimpleHashNFTs` is not called first (that's where NFTs with undefined names are removed).